pkg/metadataapi/server/middleware/auth.go

```diff
@@ -160,10 +166,21 @@ func (ka *KubernetesAuthenticator) injector(mgrs *builder.MetadataBuilder, tags
 		mgrs.SetSpec(configmap.NewSpecManager(action, immutableMap))
 		mgrs.SetState(configmap.NewStateManager(action, mutableMap))
 
+		logContext := ""
+		switch action.Type().Singular {
+		case model.ActionTypeTrigger.Singular:
+			logContext = fmt.Sprintf("tenants/%s/triggers/%s", claims.RelayTenantID, claims.RelayTriggerID)
+		case model.ActionTypeStep.Singular:
+			logContext = fmt.Sprintf("tenants/%s/runs/%s/steps/%s", claims.RelayTenantID, claims.RelayRunID, claims.RelayName)
+		}
+
+		mgrs.SetLogs(service.NewLogManager(ka.logServiceClient, logContext))
```

Comment:
If the log service client or log context ends up empty, we might not want to set this manager at all (leave it with the reject manager).

Reply:
Will do.